### ISSUE TITLE
Use mouse button as keyboard modifier key

### ISSUE THREAD

**Issue by Odrin** [linearmouse/linearmouse#582]
### OS

macOS 13.3.1 (a) (Build 22E772610a)

### LinearMouse

v0.9.3

### Describe the solution you'd like

Is it possible to use mouse additional button as a keyboard modifier keys like `shift` or `control`? I'm able to bind mouse button to keyboard shortcut (ex. `shift`), but it doesn't work as expected: holding mouse button doesn't work as pressed `shift` key modifier, but triggers single `shift` key press on release (or multiple individual button presses if I check "repeat on hold" option).

**Comment by lujjjh** [comment]
Thanks for your feedback!

Could you provide the steps to reproduce?

I tried mapping button 3 to <kbd>shift</kbd>+<kbd>command</kbd> (without checking "repeat on hold" option), and then holding down button 3 while pressing key <kbd>3</kbd>, and a screenshot was successfully captured.

**Comment by Odrin** [comment]
@lujjjh No, you got me wrong. I want to use mouse button as <kbd>shift</kbd> (or any other modifier), not as <kbd>shift</kbd> + something. I mean holding mouse button and getting UPPER case while typing.

**Comment by lujjjh** [comment]
I mapped button 3 to <kbd>shift</kbd> and it worked as expected. Do you have Logi Options+ or other mouse utilities installed?

It would be helpful to attach the logs which you can export from General Settings.

**Comment by Odrin** [comment]
Sorry for the long response. I don't have any other mouse utilities installed.
[Here](https://drive.google.com/file/d/1dSMZSDJWGcyHgxGDARkQcbCUn8QjXvZK/view?usp=drive_link) is my log. I mapped button 3 to <kbd>shift</kbd> and wrote "test" three times while holding down the mouse button 3.

**Comment by lujjjh** [comment]
Thank you for providing the logs!

Based on the logs, it appears that button 3 (identified as usage ID 04) was pressed and released within a second. This supports your hypothesis that this might be a hardware issue.

```
2023-09-09 13:34:47 +0000	info	Device	Received input value from: USB Receiver (VID=0x046D, PID=0xC52B): timestamp: 10081340369354 length: 1 data: [1] integerValue: 1 element=(usagePage: 09 usage: 04)
2023-09-09 13:34:47 +0000	info	ButtonActions	Down keys: [⇧]
2023-09-09 13:34:47 +0000	info	Device	Received input value from: USB Receiver (VID=0x046D, PID=0xC52B): timestamp: 10081340752417 length: 1 data: [0] integerValue: 0 element=(usagePage: 09 usage: 04)
2023-09-09 13:34:47 +0000	info	ButtonActions	Up keys: [⇧]
```

I would leave this issue open and see if others can reproduce it.

**Comment by RumplesWrinkles** [comment]
Hi everyone! I am having a similar issue. I have a Logitech G602, which has 11 programmable buttons.

I have assigned one of the side buttons to the "shift" keyboard shortcut, to use it with a left click. The idea is that I can hold that button, click a URL, and it will open into a new window.

For some reason, this doesn't seem to work. The button functions as a shift correctly when pressed on the mouse (letters are typed in uppercase), but when held down and used with a left click, links do not open into a new window.

Any ideas?

**Comment by vegadj** [comment]
I have the same issue.
I assigned one mouse button to act as Ctrl. What I want is to use Ctrl + mouse scroll to toggle the Accessibility Zoom feature in macOS.

macOS does detect the Ctrl key correctly — I can see it in the Virtual Keyboard. However, when I scroll the mouse wheel at the same time, the zoom feature does not activate.
The same problem happens with other combinations, such as Ctrl + right-click or Ctrl + click etc. My main aim to use one key as CTRL CMD etc and with mouse only to toggle more commands. 

My mouse is no longer supported by Logitech software on the current macOS version. This setup used to work correctly in Logitech Control Center, where modifier keys could be assigned to mouse buttons.
Has anyone found a workaround or solution for this on newer macOS versions?
